### DISCUSSION TITLE
Customize admin app snackbar error message

### DISCUSSION THREAD

**Discussion by adibfarrasy** [pocketbase/pocketbase#5583] [Q&A] [answered]
I tried to do this in my pocketbase Go app, expecting the error message to be displayed in the app.
```
package main

import (
	"log"
	"os"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/apis"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

    app.OnModelBeforeCreate().Add(func(e *core.ModelEvent) error {
        return fmt.Errorf("lorem ipsum")
    })

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```
However, I got the generic message "Failed to create record."
![image](https://github.com/user-attachments/assets/e3b1ec47-f701-49b3-bcf2-da6fa84d1741)

Is there any way I can customize the snackbar error message?

**Comment by thangngoc89** [comment]
I'm not sure about admin app specifically, but if you would like to return non-generic error from the API, you need to wrap it like this

```go
return apis.NewBadRequestError("human readable message", validation.Errors{
    // field -> error
    "machine": validation.NewError("machine_code", "machine message"),
})
```

See this for more information https://github.com/pocketbase/pocketbase/discussions/4949

**Comment by ganigeorgiev** [comment, marked as answer]
Plain errors are not surfaced to the API because they could contain sensitive information and only `apis.ApiError` are considered public safe. The caveat is that at the moment we allow returning `apis.ApiError` **only** from routes, middlewares and `*Request` related hooks; error returned from other places is wrapped as generic 400 ApiError (see https://github.com/pocketbase/pocketbase/discussions/3464; this has already changed with the refactoring in the upcoming v0.23.0 release).
